### ISSUE TITLE
Getting unable to stop activity java.lang.IndexOutOfBoundsException: Index: 3, Size: 2 when integrating Session Replay

### ISSUE THREAD

**Issue by Wintjen** [getsentry/sentry-java#3859]
### Integration

sentry-android

### Build System

Gradle

### AGP Version

8.3.0

### Proguard

Enabled

### Version

7.16.0

### Steps to Reproduce

1. Follow steps in [Sentry Session Replay setup docs](https://docs.sentry.io/platforms/android/session-replay/)
2. After adding  options.experimental.sessionReplay.onErrorSampleRate = 1.0 and  options.experimental.sessionReplay.sessionSampleRate = 1.0 starting my app
3. App shows splash screen and tries to navigate

### Expected Result

App should make it past home screen and continue into the app as normal

### Actual Result

The app crashes with the following stack trace

FATAL EXCEPTION: main
                                                                                                    Process: com.myApp, PID: 15002
                                                                                                    java.lang.RuntimeException: Unable to stop activity {com.myApp.splash.SplashScreenActivity}: java.lang.IndexOutOfBoundsException: Index: 3, Size: 2
                                                                                                    	at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:5202)
                                                                                                    	at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:5174)
                                                                                                    	at android.app.ActivityThread.handleStopActivity(ActivityThread.java:5239)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:234)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2308)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201)
                                                                                                    	at android.os.Looper.loop(Looper.java:288)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7898)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
                                                                                                    Caused by: java.lang.IndexOutOfBoundsException: Index: 3, Size: 2
                                                                                                    	at java.util.ArrayList.get(ArrayList.java:437)
                                                                                                    	at android.view.WindowManagerGlobal.setStoppedState(WindowManagerGlobal.java:658)
                                                                                                    	at android.app.Activity.performStop(Activity.java:8506)
                                                                                                    	at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:5194)
                                                                                                    	at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:5174) 
                                                                                                    	at android.app.ActivityThread.handleStopActivity(ActivityThread.java:5239) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:234) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2308) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:201) 
                                                                                                    	at android.os.Looper.loop(Looper.java:288) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7898) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)

**Comment by romtsn** [comment]
hi @Wintjen, sorry you're having troubles with replay! Could you be a bit more specific about your setup? Someone's [mentioned](https://github.com/getsentry/sentry-react-native/issues/4241) using foreground service can help reproducing the issue, is this also the case for you?

It's not reproducible in our sample app, so I'm looking for clues to help reproduce this. Do you launch your next activity from SplashScreenActivity in `onCreate`/`onStart`?

**Comment by Wintjen** [comment]
Yes in my splash screen activity, I have an onCreate that does some logic then tries to navigate me to the rest of my application. The crash is happening before this process is finished since we don't ever make it pass the splash screen. My app is working fine till trying to integrate session replay.


Here is my SplashScreenActivity.kt
```kotlin
package com.myApp.splash

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.lifecycle.LiveData
import androidx.navigation.findNavController
import androidx.navigation.fragment.NavHostFragment
import com.myApp.appcore.screen.base.extension.observeNonNull
import com.myApp.appcore.screen.common.NavigationResultKeys
import com.myApp.domain.main.TVStartingPointDomainModel
import com.myApp.R
import com.myApp.SplashScreenNavGraphDirections
import com.myApp.main.MainActivity
import dagger.hilt.android.AndroidEntryPoint

@SuppressLint("CustomSplashScreen")
@AndroidEntryPoint
class SplashScreenActivity : AppCompatActivity() {

    private val viewModel: SplashViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_splash)

        observeNonNull(viewModel.loadingDataFinished) {
            navigateToMainActivity(it, intent)
        }

        observeNonNull(viewModel.navigationEvent) {
            when (it) {
                SplashViewModel.NavigationEvent.CONNECTION_LOST -> {
                    findNavController(R.id.nav_host_fragment).navigate(
                        SplashScreenNavGraphDirections.toConnectivity()
                    )
                }
                SplashViewModel.NavigationEvent.CONNECTION_AVAILABLE -> {
                    if (viewModel.loadingDataFinished.isLoadingDataFinished().not()) {
                        viewModel.initializeData()
                    }
                }
                SplashViewModel.NavigationEvent.TO_EXIT -> {
                    findNavController(R.id.nav_host_fragment)
                        .navigate(SplashScreenNavGraphDirections.toError())
                    findNavController(R.id.nav_host_fragment)
                        .navigate(SplashScreenNavGraphDirections.toExit())
                }
            }
        }
        observeNonNull(viewModel.errorsStream) {
            findNavController(R.id.nav_host_fragment).navigate(
                SplashScreenNavGraphDirections.toError()
            )
        }

        val navHostFragment = supportFragmentManager
            .findFragmentById(R.id.nav_host_fragment) as NavHostFragment?
        val navController = navHostFragment?.navController
        navController?.currentBackStackEntry?.savedStateHandle
            ?.getLiveData<Boolean>(NavigationResultKeys.REFRESH_SCREEN_AFTER_ERROR)?.observe(this) {
                if (it)
                    viewModel.initializeData()
                else
                    viewModel.onDeclineRefresh()
            }
    }

    private fun navigateToMainActivity(model: TVStartingPointDomainModel, intent: Intent) {
        startActivity(MainActivity.buildIntent(this, model, intent))
        finish()
    }

    private fun LiveData<TVStartingPointDomainModel>.isLoadingDataFinished() = this.value is
            TVStartingPointDomainModel

    /* Start destination for splash nav graph*/
    class EmptyFragment: Fragment()
}
```

**Comment by romtsn** [comment]
hi @Wintjen unfortunately, I wasn't able to reproduce the issue in our sample app - I've tried adding a splash screen and then starting a new activity from within `onCreate`, but it was all working fine. Could you perhaps create a reproducible sample that reflects your setup better? Thanks!

**Comment by hls-app** [comment]
Got this in Production. 

```
java.lang.RuntimeException: Unable to stop activity {com.example/com.example.MainActivity}: java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 1
    at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:5906)
    at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:5872)
    at android.app.ActivityThread.handleStopActivity(ActivityThread.java:5943)
    at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:240)
    at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:205)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:177)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:98)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2685)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:230)
    at android.os.Looper.loop(Looper.java:319)
    at android.app.ActivityThread.main(ActivityThread.java:8919)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 1
    at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
    at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
    at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
    at java.util.Objects.checkIndex(Objects.java:359)
    at java.util.ArrayList.get(ArrayList.java:434)
    at android.view.WindowManagerGlobal.setStoppedState(WindowManagerGlobal.java:826)
    at android.app.Activity.performStop(Activity.java:9229)
    at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:5898)
    at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:5872)
    at android.app.ActivityThread.handleStopActivity(ActivityThread.java:5943)
    at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:240)
    at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:205)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:177)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:98)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2685)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loopOnce(Looper.java:230)
    at android.os.Looper.loop(Looper.java:319)
    at android.app.ActivityThread.main(ActivityThread.java:8919)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
```


I am using flutter and has the ` sentry_flutter: "^8.10.1"`

Just enabled the replay in the last build.

**Comment by krystofwoldrich** [comment]
Thank you for the message @hls-app

We are currently working on a fix for this issue.

**Comment by seekzeek** [comment]
Hi @krystofwoldrich we also get this crash reports in Production. Been getting these errors lately since enabling sentry-replay and quite hard to reproduce.

"@sentry/react-native": "^5.34.0"

`          Fatal Exception: java.lang.IndexOutOfBoundsException: Index 2 out of bounds for length 2
       at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
       at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
       at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
       at java.util.Objects.checkIndex(Objects.java:359)
       at java.util.ArrayList.get(ArrayList.java:434)
       at android.view.WindowManagerGlobal.updateViewLayout(WindowManagerGlobal.java:612)
       at android.view.WindowManagerImpl.updateViewLayout(WindowManagerImpl.java:157)
       at android.app.Dialog.onWindowAttributesChanged(Dialog.java:1231)
       at io.sentry.android.replay.util.FixedWindowCallback.onWindowAttributesChanged(FixedWindowCallback.java:141)
       at android.view.Window.dispatchWindowAttributesChanged(Window.java:1299)
       at com.android.internal.policy.PhoneWindow.dispatchWindowAttributesChanged(PhoneWindow.java:3251)
       at android.view.Window.setFlags(Window.java:1285)
       at android.view.Window.clearFlags(Window.java:1259)
       at com.facebook.react.views.modal.ReactModalHostView.showOrUpdate(ReactModalHostView.java:344)
       at com.facebook.react.views.modal.ReactModalHostManager.onAfterUpdateTransaction(ReactModalHostManager.java:164)
       at com.facebook.react.views.modal.ReactModalHostManager.onAfterUpdateTransaction(ReactModalHostManager.java:31)
       at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:88)
       at com.facebook.react.uimanager.ViewManager.createViewInstance(ViewManager.java:188)
       at com.facebook.react.uimanager.ViewManager.createView(ViewManager.java:115)
       at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:281)
       at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:194)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.dispatchPendingNonBatchedOperations(UIViewOperationQueue.java:1110)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1081)
       at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
       at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175)
       at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1687)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1698)
       at android.view.Choreographer.doCallbacks(Choreographer.java:1153)
       at android.view.Choreographer.doFrame(Choreographer.java:1069)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1646)
       at android.os.Handler.handleCallback(Handler.java:958)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loopOnce(Looper.java:230)
       at android.os.Looper.loop(Looper.java:319)
       at android.app.ActivityThread.main(ActivityThread.java:8919)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:578)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
        `

**Comment by RafaelPlantard** [comment]
I'm getting something similar in Sentry for Flutter on Android devices.

**Comment by krystofwoldrich** [comment]
@RafaelPlantard Thank you for the comment, could you open an issue in https://github.com/getsentry/sentry-dart with details about your app and the version of the SDK you are using, please?

**Comment by romtsn** [comment]
@RafaelPlantard (assuming you're using the latest version of the flutter SDK https://github.com/getsentry/sentry-dart/releases/tag/8.12.0)

**Comment by knro** [comment]
We are getting lots of errors with session replay as well in react-native.

```
sentry.java.android.react-native
Version	7.18.0
```

Looks like a fix been released for 7.19.0?

**Comment by romtsn** [comment]
@knro yes, RN SDK version 6.5.0 onwards should have it resolved.
